docs/traceflow-guide.md

```diff
@@ -0,0 +1,101 @@
+
+Antrea supports using Traceflow for network diagnosis.
+Creating a new Traceflow CRD triggers Traceflow module to inject packet into OVS, provide various observation points
+along the packet's path and populate these observations into status of Traceflow CRD.
+Users can start a new trace simply from either Kubectl or Antrea-Octant-Plugin and view Traceflow result via CRD or UI graph.
+We will also provide corresponding Antctl to start a new trace in the near future.
+
+## Table of Contents
+
+- [Prerequisites](#Prerequisites)
+- [Start a New Trace](#Start-a-New-Trace)
+  - [Using kubectl and YAML file](#using-kubectl-and-YAML-file)
+  - [Using Octant with antrea-octant-plugin](#Using-Octant-with-antrea-octant-plugin)
+- [View Traceflow Result and Graph](#View-Traceflow-Result-and-Graph)
+- [View Traceflow CRDs](#View-Traceflow-CRDs)
+
+## Prerequisites
+You need to switch on traceflow from featureGates defined in antrea.yml for both Controller and Agent.
+```yaml
+  antrea-controller.conf: |
+    featureGates:
+    # Enable traceflow which provides packet tracing feature to diagnose network issue.
+      Traceflow: true
+  antrea-agent.conf: |
+    featureGates:
+    # Enable traceflow which provides packet tracing feature to diagnose network issue.
+      Traceflow: true
+```
+For antrea-octant-plugin installation, please refer to [antrea-octant-installation](/docs/octant-plugin-installation.md).
+
+## Start a New Trace
+You can choose to use either Kubectl together with YAML file or Octant UI to start a new trace.
+If you use Kubectl to start a new trace, we now support traceflow from Pod to either Pod or IP and
+you can also mark the protocol(TCP/UDP/ICMP) and desired ports of the trace. 
+If you use UI to start a new trace, we currently only supports destination to be Pod, but will support
+destination to be IP or Service soon.
+
+### Using kubectl and YAML file
+You can start a new trace by creating Traceflow CRD via Kubectl and a YAML file which contains the essential
+configuration of Traceflow CRD. An example YAML file of Traceflow CRD might look like this:
+```yaml
+apiVersion: ops.antrea.tanzu.vmware.com/v1alpha1
+kind: Traceflow
+metadata:
+  name: tf-test
+spec:
+  source:
+    namespace: default
+    pod: tcp-sts-0
+  destination:
+    namespace: default
+    pod: tcp-sts-2
+#   ip: IP can also be marked as destination
+  packet:
+    ipHeader:
+      protocol: 6 # Protocol here can be 6 (TCP), 17 (UDP) or 1 (ICMP), default value is 1 (ICMP)
+    transportHeader:
+      tcp:
+        srcPort: 10000 # Source port needs to be set when Protocol is TCP/UDP.
+        dstPort: 80 # Destination port needs to be set when Protocol is TCP/UDP.
+```
+The CRD above starts a new trace from port 10000 of source Pod named tcp-sts-0 to port 80
+of destination Pod named tcp-sts-2 using TCP protocol.
+
+### Using Octant with antrea-octant-plugin
+
+<img src="http://downloads.antrea.io/static/tf_create.png" width="600" alt="Start a New Trace">
```

Comment:
please use the full S3 link so that images can be loaded over https. It doesn't matter that the link is long since users won't be using the link directly